modules/apps/dispatch/dispatch-test/src/testIntegration/java/com/liferay/dispatch/internal/messaging/test/SchedulerResponseTest.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dispatch.internal.messaging.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.portal.kernel.messaging.Message;
+import com.liferay.portal.kernel.model.CompanyConstants;
+import com.liferay.portal.kernel.scheduler.SchedulerEngineHelper;
+import com.liferay.portal.kernel.scheduler.SchedulerException;
+import com.liferay.portal.kernel.scheduler.StorageType;
+import com.liferay.portal.kernel.scheduler.messaging.SchedulerResponse;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Luis Ortiz
+ */
+@RunWith(Arquillian.class)
+public class SchedulerResponseTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testStartupScheduleJobCompanyId() throws SchedulerException {
+		SchedulerResponse schedulerResponse =
```

Comment:
@LuisOrtiz89 Scheduled jobs itself has nothing related to company, it is that when a job is triggered, it includes the company context. You can not test like this as next time the company id may be changed, it depends on the runtime context

Reply:
Hi @tinatian, for what we saw, when some MessageListener are activated they create a trigger and register an entry in the scheduler. You can see an example [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-discount-service/src/main/java/com/liferay/commerce/discount/internal/messaging/CheckCommerceDiscountMessageListener.java).

In those cases, when the scheduled job is triggered the message comes populated with a company ID that corresponds with the company ID set in the CompanyThreadLocal at the moment of the component activation. 

We need that this company ID is SYSTEM, as it indicates that the message is not directed to any company in particular, so the [DBPartitionMessageBusInterceptor](https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal/portal-db-partition/src/main/java/com/liferay/portal/db/partition/internal/messaging/DBPartitionMessageBusInterceptor.java#L53) can generate as many messages as company IDs are registered.

Reply:
Hi @tinatian, is everything clear? Did you were able to take another look at it?

Reply:
Hi @tinatian, as discussed, I removed the test and changed the commit and PR message.
You can see the previously QA review in this PR: https://github.com/liferay-uniform/liferay-portal/pull/869